partiql-conformance-test-generator/src/generator.rs

```diff
@@ -0,0 +1,63 @@
+use crate::schema::TestCaseKind::Parse;
+use crate::schema::{Namespace, ParseAssertions, ParseTestCase, TestCase, TestDocument};
+use codegen::{Function, Module, Scope};
+
+/// Defines a test code generation object
+pub struct Generator {
+    pub test_document: TestDocument,
+}
+
+impl Generator {
+    /// Generates a `Scope` from the `Generator`'s `test_document`
+    pub fn generate_scope(&self) -> Scope {
+        test_document_to_scope(&self.test_document)
+    }
+}
+
+/// Converts a `TestDocument` into a `Scope`
+fn test_document_to_scope(test_document: &TestDocument) -> Scope {
+    let mut scope = Scope::new();
+    for namespace in &test_document.namespaces {
+        scope.push_module(namespace_to_module(namespace));
+    }
+    for test in &test_document.test_cases {
+        scope.push_fn(test_case_to_function(test));
+    }
+    scope
+}
+
+/// Converts a `Namespace` into a `Module`
+fn namespace_to_module(namespace: &Namespace) -> Module {
+    let mut module = Module::new(&*namespace.name);
+    for ns in &namespace.namespaces {
+        module.push_module(namespace_to_module(ns));
+    }
+    for test in &namespace.test_cases {
+        module.push_fn(test_case_to_function(test));
+    }
+    module
+}
+
+/// Converts a test case into a testing `Function`
+fn test_case_to_function(test_case: &TestCase) -> Function {
+    match &test_case.test_kind {
+        Parse(ParseTestCase { parse_assertions }) => {
+            let mut test_fn: Function = Function::new(&test_case.test_name);
+            test_fn.attr("test").line(format!(
+                "let parse_result = partiql_parser::parse_partiql(\"{}\");",
```

Comment:
Discovered in some tests I added to `partiql-tests` that I need to escape the quotes here. Possibly using a [raw string literal](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals). Will fix in an upcoming commit.

Reply:
Fixed in https://github.com/partiql/partiql-lang-rust/pull/75/commits/b5ab2b2ccf960636ee232d0fcebfe0ae52565a59